### PR TITLE
Update tx with events after execution

### LINKED ISSUE #137: wrong documentation for `calculate_event_commitment` function
```rust
/// Calculate transaction commitment hash value.
///
/// The transaction commitment is the root of the Patricia Merkle tree with height 64
/// constructed by adding the (transaction_index, transaction_hash_with_signature)
/// key-value pairs to the tree and computing the root hash.
///
/// # Arguments
///
/// * `transactions` - The transactions to get the root from.
///
/// # Returns
///
/// The merkle root of the merkle tree built from the transactions.
pub fn calculate_event_commitment<T: CryptoHasher>(transactions: &[Transaction]) -> (H256, u128) {
    let mut tree = CommitmentTree::<T>::default();
    let mut len = 0_u64;
    transactions.iter().flat_map(|tx| tx.events.iter()).for_each(|event| {
        len += 1;
        let final_hash = calculate_event_hash::<T>(event);
        tree.set(len - 1, final_hash);
    });
    (H256::from_slice(&tree.commit().to_bytes_be()), len as u128)
}
```

### PR DESCRIPTION
**Pull request by LucasLvy**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [x] Documentation content changes
- [x] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: Fixes #137 

# What is the new behavior?

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

# Other information

### CODE REVIEW

File: crates/pallets/starknet/src/lib.rs [hunk 1]
```diff
@@ -623,27 +629,39 @@ pub mod pallet {
...
-                        }
-                        Err(e) => {
-                            log!(error, "Error parsing event: {:?}", e);
-                            return Err(e);
-                        }
-                    }
-                }
-            }
+        #[inline(always)]
+        fn emit_events(call_info: &mut CallInfo, transaction: &mut Transaction) -> Result<(), EventError> {
```
Comment (abdelhamidbakhta) [inline crates/pallets/starknet/src/lib.rs:633]: add rust doc

File: crates/pallets/starknet/src/lib.rs [hunk 2]
```diff
@@ -623,27 +629,39 @@ pub mod pallet {
...
+        /// # Error
+        ///
+        /// Returns an error if the event construction fails.
+        #[inline(always)]
+        fn emit_event(
+            event: &EventContent,
+            from_address: ContractAddress,
+            transaction: &mut Transaction,
+        ) -> Result<(), EventError> {
+            log!(info, "Transaction event: {:?}", event);
```
Comment (abdelhamidbakhta) [inline crates/pallets/starknet/src/lib.rs:658]: switch to debug or even trace

File: crates/primitives/starknet/src/execution/mod.rs [hunk 3]
```diff
@@ -72,6 +72,16 @@ impl From<ContractClass> for ContractClassWrapper {
         }
     }
 }
+impl From<&starknet_api::state::ContractClass> for ContractClassWrapper {
+    fn from(contract_class: &starknet_api::state::ContractClass) -> Self {
```
Comment (abdelhamidbakhta) [inline crates/primitives/starknet/src/execution/mod.rs:76]: add rust doc

**Review by abdelhamidbakhta — CHANGES_REQUESTED** [review]
left few comments

File: crates/pallets/starknet/src/lib.rs (hunk 1 above)
Comment (LucasLvy) [reply crates/pallets/starknet/src/lib.rs:633]: The rust doc is already there

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

**Pull request merged**